### PR TITLE
#160577597 Upgrade from Django 1.10 to Django 2.1

### PR DESCRIPTION
**Pull request by bevkololi**
##What does this PR do?
Has updates on upgrading the authors haven django project from Django 1.10 to Django 2.1
#### Description of Task to be completed?
The code base was created using Django 1.10 and as per the requirements this new one uses Django 2.10
#### How should this be manually tested?
By running the command "python" Followed by "Import django" and finally "django.VERSION"
#### What are the relevant pivotal tracker stories?
160577597

### CODE REVIEW

**Comment by YAHYA-H** [comment]
🎆 🔥 🍾 💯 , Awesome!! 👍

**Comment by YAHYA-H** [comment]
💯 🔥 AWESOME!!!
@bevkololi change your base to "develop" 😄

**Comment by gitaumoses4** [comment]
@bevkololi you can also change the *url* schema to *path*, as part of the upgrade to Django 2.1.
All is well on the settings and requirements.

**Comment by YAHYA-H** [comment]
💯 🔥 AWESOME!!!
@bevkololi change your base to "develop" 😄

File: .gitignore [hunk 1]
```diff
@@ -91,3 +91,7 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
+.DS_Store
+
+/migrations
```
Comment (esirK) [inline .gitignore]: I think we should be pushing the migrations. Kindly remove it from this file.

File: authors/apps/authentication/admin.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from django.contrib import admin
```
Comment (esirK) [inline authors/apps/authentication/admin.py]: Is this file related to the migration from 1.10 to 2.1??

File: authors/apps/authentication/urls.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
     LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView
 )
 
+app_name = "authentication"
+
 urlpatterns = [
     url(r'^user/?$', UserRetrieveUpdateAPIView.as_view()),
```
Comment (esirK) [inline authors/apps/authentication/urls.py]: as suggested in a comment, use `path` instead of `url`.

File: authors/settings.py [hunk 4]
```diff
@@ -136,8 +136,42 @@
 REST_FRAMEWORK = {
     'EXCEPTION_HANDLER': 'authors.apps.core.exceptions.core_exception_handler',
     'NON_FIELD_ERRORS_KEY': 'error',
-
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
+        'rest_framework_jwt.authentication.JSONWebTokenAuthentication',
```
Comment (esirK) [inline authors/settings.py]: Are this changes related to the migration from 1.10 to 2.1 ??

File: .gitignore (hunk 1 above)
Comment (SrMoffat) [reply .gitignore]: I was wondering whether or not the migrations should be included. Thanks for clarifying it.

File: authors/apps/authentication/admin.py (hunk 2 above)
Comment (SrMoffat) [reply authors/apps/authentication/admin.py]: The reason why it was included was to give people access to the browsable version of the app so we can check whether the admin route works. It was not meant to be pushed so this was a slight error that will be fixed.

File: authors/settings.py (hunk 4 above)
Comment (SrMoffat) [reply authors/settings.py]: These were also some of the changes that were applied to test whether the application is working (as it was originally). It was a debugging attempt that was meant to make the app run. It was not meant to be pushed, we shall be vigilant from here onwards.

**Pull request merged**